### PR TITLE
protocol upgrade (v2): Burn slashed rewards (phase RD 1) + changes to system epoch info event

### LINKED ISSUE #3123: [gas-price-modulation-tf]: Burn slashed rewards
In this phase, the grading and reporting of validators remain unchanged:

1. Validators report non-behaving peers
2. Those reports are aggregated, and the set of slashed validators is computed using a fixed maximum threshold for the stake of reporters

Instead of adjusting rewards so that slashed rewards are assigned to non-slashed validators, we remove this adjustment, entirely burning the slashed rewards, which would go to the protocol instead of going to the non-slashed validator set. Even though this might seem to be a simplistic change, it has a significant impact on the incentivization properties of the protocol. Now, validators don’t have any incentive to attack or mis-report their peers, which is an undesirable property present in both SUI and IOTA Rebased.

### PR DESCRIPTION
**Pull request by oliviasaa**
# Description of change

- Changes in the reward distribution so the slashed rewards from a validator are not redistributed to the others.
  **Phase RD 1**: In this phase, the grading and reporting of validators remain unchanged. Instead of adjusting rewards so that slashed rewards are assigned to non-slashed validators, we remove this adjustment, entirely burning the slashed rewards, which would go to the protocol instead of going to the non-slashed validator set. 
- Change the information about burned tokens amount in the `SystemEpochInfoEventV1` (emitted from the advance_epoch function)

## Links to any relevant issues

Fixes #3123 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

The reward amounts were tested and checked for both slashed and non-slashed validators

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: crates/iota-framework/packages/iota-system/sources/validator_set.move [hunk 1]
```diff
@@ -341,28 +341,15 @@ module iota_system::validator_set {
...
-                get_validator_indices(&self.active_validators, &slashed_validators),
-                reward_slashing_rate,
-                &unadjusted_staking_reward_amounts,
-            );
-
-        // Compute the adjusted amounts of stake each validator should get given the tallying rule
-        // reward adjustments we computed before.
-        // `compute_adjusted_reward_distribution` must be called before `distribute_reward` and `adjust_stake_and_gas_price` to
-        // make sure we are using the current epoch's stake information to compute reward distribution.
+       // Compute the adjusted amounts of stake each validator should get given the tallying rule
```
[the remark this answers is not in the document]
Comment (cyberphysic4l) [reply crates/iota-framework/packages/iota-system/sources/validator_set.move]: Yes, the comment is still correct, we will still refer to the calculation as the "tallying rule" even though we have changed it, and we still pass information about which validators are slashed and the slashing rate.

**Review by cyberphysic4l — APPROVED** [review]
LGTM

File: crates/iota-open-rpc/spec/openrpc.json [hunk 2]
```diff
@@ -1300,7 +1300,7 @@
             "name": "Result",
             "value": {
               "minSupportedProtocolVersion": "1",
-              "maxSupportedProtocolVersion": "1",
+              "maxSupportedProtocolVersion": "2",
               "protocolVersion": "1",
```
Comment (Thoralf-M) [inline crates/iota-open-rpc/spec/openrpc.json:1304]: Shouldn't this one also be bumped?

Comment (miker83z) [reply crates/iota-open-rpc/spec/openrpc.json:1304]: It's an example, so I don't think so...

Comment (Thoralf-M) [reply crates/iota-open-rpc/spec/openrpc.json:1304]: Good point, didn't look further up

**Review by valeriyr — APPROVED** [review]
LGTM!

**Review by lzpap — APPROVED** [review]
LGTM!

**Review by kodemartin — APPROVED** [review]
lgtm

File: crates/iota-protocol-config/src/lib.rs [hunk 3]
```diff
@@ -16,11 +16,13 @@ use tracing::{info, warn};
...
 /// The minimum and maximum protocol versions supported by this build.
 const MIN_PROTOCOL_VERSION: u64 = 1;
-pub const MAX_PROTOCOL_VERSION: u64 = 1;
+pub const MAX_PROTOCOL_VERSION: u64 = 2;
 
 // Record history of protocol version allocations here:
 //
 // Version 1: Original version.
+// Version 2: Don't redistribute slashed staking rewards, fix computation of
+// SystemEpochInfoEventV1
```
Comment (thibault-martinez) [inline crates/iota-protocol-config/src/lib.rs]:
```suggestion
// SystemEpochInfoEventV1.
```

File: crates/iota-protocol-config/src/lib.rs [hunk 4]
```diff
@@ -1159,8 +1161,8 @@ impl ProtocolConfig {
         #[cfg(msim)]
         {
             // populate the fake simulator version # with a different base tx cost.
-            if version == ProtocolVersion::MAX_ALLOWED {
-                let mut config = Self::get_for_version_impl(version - 1, Chain::Unknown);
+            if version > ProtocolVersion::MAX {
+                let mut config = Self::get_for_version_impl(ProtocolVersion::MAX, Chain::Unknown);
```
Comment (thibault-martinez) [inline crates/iota-protocol-config/src/lib.rs:1166]: Just making sure the change from `MAX_ALLOWED` to `MAX` is intended as we're in a `msim` block and `MAX_ALLOWED` has a different value depending on `msim`

**Review by thibault-martinez — APPROVED** [review]

File: crates/iota-protocol-config/src/lib.rs (hunk 4 above)
Comment (valeriyr) [reply crates/iota-protocol-config/src/lib.rs:1166]: Yes, that's right. For the `msim` configuration, `MAX_ALLOWED` is changed to contain extra fake versions which are used in tests.
In Sui it contains 1 extra version, so we needed to create a config for `version - 1`.
In our case, we have several extra versions, so we need to change this logic to create a configuration.
P.S. It is impossible to create a configuration for a version bigger than `MAX`.

Comment (thibault-martinez) [reply crates/iota-protocol-config/src/lib.rs:1166]: Thank you for the confirmation

**Pull request merged**